modules/mono/glue/Managed/Files/StringExtensions.cs

```diff
@@ -97,6 +97,126 @@ public static string[] Bigrams(this string instance)
 
+        // <summary>
+        // Return the amount of substrings in string. Case-sensitive.
+        // </summary>
+        public static int Count(this string instance, string substr, int from = 0, int to = 0)
+        {
+            if (substr.Length == 0)
+            {
+                return 0;
+            }
+
+            int len = instance.Length;
+            int slen = substr.Length;
+
+            if (len < slen)
+            {
+                return 0;
+            }
+
+            string str;
+
+            if (from >= 0 && to >= 0)
+            {
+                if (to == 0)
+                {
+                    to = len;
+                }
+                else if (from >= to)
+                {
+                    return 0;
+                }
+                if (from == 0 && to == len)
+                {
+                    str = instance;
+                }
+                else
+                {
+                    str = instance.Substring(from, to - from);
+                }
+            }
+            else
+            {
+                return 0;
+            }
+
+            int c = 0;
+            int idx;
+
+            do
+            {
+                idx = str.IndexOf(substr, StringComparison.Ordinal);
+                if (idx != -1)
+                {
+                    str = str.Substring(idx + slen);
+                    ++c;
+                }
+            } while (idx != -1);
+
+            return c;
+        }
+
+        // <summary>
+        // Return the amount of substrings in string. Case-insensitive.
+        // </summary>
+        public static int CountN(this string instance, string substr, int from = 0, int to = 0)
+        {
+            if (substr.Length == 0)
+            {
+                return 0;
+            }
+
+            int len = instance.Length;
+            int slen = substr.Length;
+
+            if (len < slen)
+            {
+                return 0;
+            }
+
+            string str;
+
+            if (from >= 0 && to >= 0)
+            {
+                if (to == 0)
+                {
+                    to = len;
+                }
+                else if (from >= to)
+                {
+                    return 0;
+                }
+                if (from == 0 && to == len)
+                {
+                    str = instance;
+                }
+                else
+                {
+                    str = instance.Substring(from, to - from);
+                }
+            }
+            else
+            {
+                return 0;
+            }
+
+            int c = 0;
+            int idx;
+
+            do
+            {
+                idx = str.IndexOf(substr, StringComparison.OrdinalIgnoreCase);
+                if (idx != -1)
+                {
+                    str = str.Substring(idx + slen);
+                    ++c;
+                }
+            } while (idx != -1);
+            
+            return c;
+        }
+
```

Comment:
I would prefer this as a single function with the case sensitivity as an optional parameter.

Reply:
Fixed